build_docs.pl

```diff
@@ -426,17 +457,22 @@ sub init_repos {
         say "Removing old repo <" . $dir->basename . ">";
         $dir->rmtree;
     }
-    return $repos_dir;
+    return ($repos_dir, $temp_dir, $target_repo, $target_repo_checkout);
 }
 
 #===================================
 sub push_changes {
 #===================================
-    my $build_dir = shift;
+    my ($build_dir, $target_repo, $target_repo_checkout) = @_;
 
+    local $ENV{GIT_WORK_TREE} = $target_repo_checkout if $target_repo;
+    local $ENV{GIT_DIR} = $ENV{GIT_WORK_TREE} . '/.git' if $target_repo;
+
+    say 'Building revision.txt';
```

Comment:
I saw these steps take a while when I tested locally so I added this so I could watch.